src/vs/workbench/contrib/search/common/searchModel.ts

```diff
@@ -225,14 +226,14 @@ export class FileMatch extends Disposable implements IFileMatch {
 		private _closestRoot: FolderMatchWorkspaceRoot | null,
 		@IModelService private readonly modelService: IModelService,
 		@IReplaceService private readonly replaceService: IReplaceService,
-		@ILabelService private readonly labelService: ILabelService,
+		@ILabelService readonly labelService: ILabelService,
 	) {
 		super();
 		this._resource = this.rawMatch.resource;
 		this._matches = new Map<string, Match>();
 		this._removedMatches = new Set<string>();
 		this._updateScheduler = new RunOnceScheduler(this.updateMatchesForModel.bind(this), 250);
-
+		this._name = labelService.getUriBasenameLabel(this.resource);
```

Comment:
If this is expensive, you can delay doing this work until it's needed. We have a helper, `Lazy`, to do this for you